compiler/rustc_mir_build/src/thir/cx/expr.rs

```diff
@@ -395,7 +396,21 @@ fn make_mirror_unadjusted<'a, 'tcx>(
                 .zip(substs.upvar_tys())
                 .map(|(captured_place, ty)| capture_upvar(cx, expr, captured_place, ty))
                 .collect();
-            ExprKind::Closure { closure_id: def_id, substs, upvars, movability }
+
+            // Then introduce fake reads based on what we collected during typeck
+            debug!("RFC2229 = closure_fake_read {:?}", cx.typeck_results().closure_fake_reads);
+            let fake_reads = match cx.typeck_results().closure_fake_reads.get(&def_id) {
+                Some(vals) => vals.clone(),
+                None => Vec::<HirPlace<'tcx>>::new(),
```

Comment:
you pronbably want to convert this to ExprRef? Look at `capture_upvar` or `convert_upvar`

Reply:
Essentially convert HirPlace in terms to ThirExpr in the  parent's context